### PR TITLE
Kindred Spirit

### PR DESCRIPTION
**Pull request by Rocketeer456**
## Summary
A four-part mission where you befriend a fellow rookie pilot over time. The final reward is making him a permanent escort.

## Save File
Required: >500000 credits, >100 combat rating
[Test Pilot~Kindred Spirit 1.txt](https://github.com/endless-sky/endless-sky/files/9191137/Test.Pilot.Kindred.Spirit.1.txt)
[Test Pilot~Kindred Spirit 2 Correction.txt](https://github.com/endless-sky/endless-sky/files/9191138/Test.Pilot.Kindred.Spirit.2.Correction.txt)
[Test Pilot~Kindred Spirit 3 Correction.txt](https://github.com/endless-sky/endless-sky/files/9191141/Test.Pilot.Kindred.Spirit.3.Correction.txt)
[Test Pilot~Kindred Spirit 4.txt](https://github.com/endless-sky/endless-sky/files/9191142/Test.Pilot.Kindred.Spirit.4.txt)

### CODE REVIEW

File: data/human/human missions.txt [hunk 1]
```diff
@@ -4175,3 +4175,322 @@ mission "Hai Wormhole Warning"
...
+				`	"I'd love to have you as a member of my fleet."`
+					goto yep
+				`	"I think you should carve your own story, Charlie. I wouldn't want to hold you back."`
+					goto nah
+			label yep
+			`	"Heck yeah! I'll get my things!" You wait as he rushes around his home, quickly throwing together a suitcase of hopefully clean laundry. His mom makes you promise you'll take care of her son, especially against pirates. Soon, you and Charlie the train back to the spaceport together."`
+				accept
+			label nah
+			`	"Oh, stop it," he blushes. "Well, it's been fun. I hope your adventures are long and fun. You deserve it."`
+				decline
```
Comment (warp-core) [inline data/human/human missions.txt:4459]:
```suggestion
			choice
				`	"I'd love to have you as a member of my fleet."`
				`	"I think you should carve your own story, Charlie. I wouldn't want to hold you back."`
					goto nah
			`	"Heck yeah! I'll get my things!" You wait as he rushes around his home, quickly throwing together a suitcase of hopefully clean laundry. His mom makes you promise you'll take care of her son, especially against pirates. Soon, you and Charlie are on the train back to the spaceport together.`
				accept
			label nah
			`	"Oh, stop it," he blushes. "Well, it's been fun. I hope your adventures are long and fun. You deserve it."`
				decline
```

File: data/human/human missions.txt [hunk 2]
```diff
@@ -4175,3 +4175,322 @@ mission "Hai Wormhole Warning"
...
+	to offer
+		has "Kindred Spirit 4: done"
+		random < 50
+	npc save
+		government "Merchant"
+		personality heroic escort
+		ship "Quicksilver" "Tyran the Brave"
+	on offer
+		conversation
+			`	"Hey, <first>!" Charlie calls. "You want to travel together again?"`
```
Comment (warp-core) [inline data/human/human missions.txt:4488]: This should probably have a little more to it. Something about you wandering through the spaceport or similar when your comm buzzes or beeps or something with a call from Charlie. As it is, it sounds like he calls immediately as we enter the spaceport, which is kinda creepy.
Or, perhaps give us the option to contact him.
`You remember this is the planet where Charlie, a young pilot you've escorted in the past and who previously joined you on your travels lives. Contact him and see if he is interested in travelling together again?`

File: data/human/human missions.txt [hunk 3]
```diff
@@ -4175,3 +4175,322 @@ mission "Hai Wormhole Warning"
...
+				`	"Sure, let's head out!"`
+					goto yep
+				`	"I'm going solo right now."`
+					goto nope
+			label yep
+				`	"Alright, I'll get my things packed. Let me know when we're taking off!"`
+					accept
+			label nope
+				`	"Alright. If you ever change your mind, check back around here. See you!"`
+					decline
```
Comment (warp-core) [inline data/human/human missions.txt:4499]: Asking us to let him know before we leave makes it sound like we need to do something, but we don't.
```suggestion
			choice
				`	"Sure, let's head out!"`
				`	"I'm going solo right now."`
					goto nope
			`	"Alright, I'll get my things packed. I'll be ready by the time you leave!"`
				accept
			label nope
			`	"Alright. If you ever change your mind, check back around here. See you!"`
				decline
```

**Review by warp-core — COMMENTED** [review]
There's a lot of unnecessary `goto`s in this.
Also, endpoints like `accept` and `decline` can still be used in `on complete` conversations, they still end the conversation.

**Comment by roadrunner56** [comment]
@Rocketeer456 Seeing as this PR has been labelled Waiting on OP for several months now, if we don't receive a response in a week this PR will be considered orphaned and closed.

**Comment by roadrunner56** [comment]
This PR hasn't been touched for months now, so closing it.

**Pull request closed without merging**